standardtracer/propagation.go

```diff
@@ -44,9 +45,11 @@ func (p *splitTextPropagator) InjectSpan(
 		fieldNameSampled: strconv.FormatBool(sc.raw.Sampled),
 	}
 	sc.Lock()
-	splitTextCarrier.TraceAttributes = make(map[string]string, len(sc.raw.Attributes))
-	for k, v := range sc.raw.Attributes {
-		splitTextCarrier.TraceAttributes[k] = v
+	if l := len(sc.raw.Attributes); l > 0 {
+		splitTextCarrier.TraceAttributes = make(map[string]string, l)
```

Comment:
There's no existing code that does this, at least not in the Go code, but I think that's because it's not possible (request headers aren't exposed as a map). It makes some sense for languages like Python, so it seems ok to define it that way.